PhotoMap/Features/Timeline/Coordinators/TimelineCoordinator.swift

```diff
@@ -52,10 +56,18 @@ class TimelineCoordinator: Coordinator {
     
     private func presentCategoryScreen() {
         let coordinator = CategoryCoordinator(diContainer: diContainer)
+        coordinator.parentCoordinator = self
         let categoryNavigationVC = coordinator.start()
         categoryNavigationVC.modalPresentationStyle = .fullScreen
         navigationController.present(categoryNavigationVC, animated: true)
         childCoordinators.append(coordinator)
     }
     
+    func childDidFinish(_ childCoordinator: Coordinator) {
+        if let index = childCoordinators.firstIndex(where: { coordinator -> Bool in
+            return childCoordinator === coordinator
+        }) {
+            childCoordinators.remove(at: index)
+        }
```

Comment:
You may use [removeAll(where:)](https://developer.apple.com/documentation/swift/array/3017530-removeall)